ebpf/throughput/main.go

```diff
@@ -0,0 +1,206 @@
+package throughput
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/cilium/ebpf/ringbuf"
+	"github.com/cilium/ebpf/rlimit"
+	"github.com/ddosify/alaz/k8s"
+	"github.com/ddosify/alaz/log"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ThroughputEventBpf struct {
+	Timestamp uint64
+	Size      uint32
+	SPort     uint16
+	DPort     uint16
+	SAddr     [16]byte
+	DAddr     [16]byte
+}
+
+// for user space
+type ThroughputEvent struct {
+	Timestamp uint64
+	Size      uint32
+	SPort     uint16
+	DPort     uint16
+	SAddr     string
+	DAddr     string
+}
+
+// $BPF_CLANG and $BPF_CFLAGS are set by the Makefile.
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc $BPF_CLANG -cflags $BPF_CFLAGS bpf throughput.c -- -I../headers
+
+const THROUGHPUT_EVENT = "throughput_event"
+
+var NODE_NAME = os.Getenv("NODE_NAME")
+
+func (e ThroughputEvent) Type() string {
+	return THROUGHPUT_EVENT
+}
+
+// returns when the program is loaded
+func DeployAndWait(ctx context.Context, ch chan interface{}, eventChan <-chan interface{}) {
+	// Allow the current process to lock memory for eBPF resources.
+	if err := rlimit.RemoveMemlock(); err != nil {
+		log.Logger.Fatal().Err(err).Msg("failed to remove memlock limit")
+	}
+
+	// Load pre-compiled programs and maps into the kernel.
+	objs := bpfObjects{}
+	if err := loadBpfObjects(&objs, nil); err != nil {
+		log.Logger.Fatal().Err(err).Msg("loading objects")
+	}
+	defer objs.Close()
+
+	ticker := time.NewTicker(time.Millisecond)
+	defer ticker.Stop()
+
+	time.Sleep(3 * time.Second)
```

Comment:
nit: @g-linville could you add a small comment that outlines why we're sleeping here, and what happens when we don't?

Reply:
There is a one second sleep here in the other metrics but I increased this one to 3 during my testing to see if it would affect anything. I'm gonna remove it now and see if that has any impact. I don't think it will